src/main/java/Duke.java

```diff
@@ -1,10 +1,77 @@
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.Scanner;
+/**
+ * Represents a todo manager bot.
+ */
 public class Duke {
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Constructs a Duke bot.
+     * @param filePath the path of the file which store the previous data.
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.readFile());
+        } catch (IOException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    /**
+     * Starts the todo list feature of the .
+     */
+    public void run() {
+        Scanner sc = new Scanner(System.in);
+        String res;
+        ui.welcome();
+        while (true) {
+            try {
+                res = sc.nextLine();
+                if (res.equalsIgnoreCase(Operations.BYE.name())) {
+                    break;
+                } else if (res.equalsIgnoreCase(Operations.LIST.name())) {
+                    ui.printList(tasks.getData());
+                } else if (res.startsWith(Operations.DONE.name().toLowerCase())) {
+                    tasks.done(res, ui, storage);
+                } else if (res.startsWith(Operations.DELETE.name().toLowerCase())) {
+                    tasks.delete(res, ui, storage);
+                } else if (res.startsWith(Operations.TODO.name().toLowerCase())) {
+                    tasks.todo(res, ui, storage);
+                } else if (res.startsWith(Operations.DEADLINE.name().toLowerCase())) {
+                    tasks.deadline(res, ui, storage);
+                } else if (res.startsWith(Operations.EVENT.name().toLowerCase())) {
+                    tasks.event(res, ui, storage);
+                } else if (res.startsWith(Operations.FIND.name().toLowerCase())) {
+                    tasks.find(res, ui);
+                } else {
+                    // Exception: eg. ???
+                    throw new DukeException("     ☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
```

Comment:
Could perhaps be create a command class to encapsulate all the commands to adhere to the OOP concept.